auth/views.py

```diff
@@ -0,0 +1,150 @@
+        if not next and self.extra_context and "next" in self.extra_context:
+            next = self.extra_context.get("next", "")
+        return f"{url}?next={next}"
+
+
+@login_required
+def mon_compte(request):
+    return render(request, "compte/index.html")
+
+
+@login_required
+def mon_identifiant(request):
+    if request.method == "POST":
+        form = forms.UsernameChangeForm(request.POST)
+        if form.is_valid():
+            username = form.cleaned_data["username"]
+            user = request.user
+            old_username = user.username
+            user.username = username
+            user.save()
+            LogEntry.objects.log_action(
+                user_id=user.id,
+                content_type_id=ContentType.objects.get_for_model(user).pk,
+                object_id=user.id,
+                object_repr=username,
+                action_flag=CHANGE,
+                change_message=f"Changement de nom d'utilisateur (avant: {old_username})",
+            )
+            messages.add_message(
+                request,
+                messages.SUCCESS,
+                f"Votre nom d'utilisateur a été changé en {user.username}.",
+            )
+            return redirect("mon_identifiant")
+    else:
+        form = forms.UsernameChangeForm(initial={"username": request.user.username})
+    return render(
+        request,
+        "compte/mon_identifiant.html",
+        context={"form": form},
+    )
+
+
+@login_required
+def mon_email(request):
+    if request.method == "POST":
+        form = forms.EmailChangeForm(request.POST)
+        if form.is_valid():
+            new_email = form.cleaned_data
+            user = request.user
+
+            activation_token = create_token(user, new_email)
+            send_activation_mail(activation_token, new_email, user)
+
+            LogEntry.objects.log_action(
+                user_id=request.user.id,
+                content_type_id=ContentType.objects.get_for_model(user).pk,
+                object_id=user.id,
+                object_repr=new_email,
+                action_flag=CHANGE,
+                change_message=f"Demande de changement d'email {user.email} -> {new_email}",
+            )
+            messages.add_message(
+                request,
+                messages.INFO,
+                f"Un email de validation vous a été envoyé à {user.email}. Merci de consulter votre boite de réception",
+            )
+            return redirect("mon_email")
+    else:
+        form = forms.EmailChangeForm(initial={"email": request.user.email})
+    return render(
+        request,
+        "compte/mon_email.html",
+        context={"form": form},
+    )
+
+
+def change_email(request, activation_token):
+    if not activation_token:
+        return redirect("mon_email")
+
+    user, failure = validate_from_token(activation_token)
+    if failure:
+        render(
+            request,
+            "compte/change_activation_failed.html",
```

Comment:
Nit: The template filename should be more explicit, eg. `email_change_activation_failed.html`